src/swagger/specGenerator2.ts

```diff
@@ -62,7 +63,16 @@ export class SpecGenerator2 extends SpecGenerator {
         };
 
         if (referenceType.additionalProperties) {
-          definitions[referenceType.refName].additionalProperties = this.buildAdditionalProperties(referenceType.additionalProperties);
+            definitions[referenceType.refName].additionalProperties = this.buildAdditionalProperties(referenceType.additionalProperties);
+        } else {
+            // Since additionalProperties was not explicitly set in the TypeScript interface for this model
+            //      ...we need to make a decision
+            if (config.noImplicitAdditionalProperties) {
```

Comment:
couldn't this just be `definitions[referenceType.refName].additionalProperties = !config.noImplicitAdditionalProperties`?

Reply:
this makes me think.. why do we use two different names that we have to be careful with because they're opposites?

Reply:
Very good point on the opposite name issue. I elected to choose a “negative” variable name because the config is optional.

The alternative would be more onerous. Like if we called it `allowAdditionalProperties` then the default would have to be true since that’s what Swagger does.

So our internal code would be:
```ts
schema.additionalProperties = config.allowAdditionalProperties !== undefined && config.allowAdditionalProperties === false ? false : true;
```

Actually as I’m writing this, it’s not that bad. Maybe I was just overthinking it? I’ll ask the people who first asked for this feature— What do you think @mushketyk and @jorshali?

Reply:
Looks like we're only waiting for people to agree on a name for the new option, otherwise this could get merged.

@endor I just pushed the remaining fixed.

Reply:
Alright, since we haven't heard any of the tie breakers, I'm going to go with my gut and merge this. There are a couple of PRs in the pipeline that depend on the structure of this code, so I'm going to merge it.